developer-docs-site/static/scripts/prover_setup.sh

```diff
@@ -0,0 +1,365 @@
+     echo "Z3 ${Z3_VERSION} already installed"
+     return
+  fi
+  if [[ "$(uname)" == "Linux" ]]; then
+    Z3_PKG="z3-$Z3_VERSION-x64-glibc-2.31"
+  elif [[ "$(uname)" == "Darwin" ]]; then
+    Z3_PKG="z3-$Z3_VERSION-x64-osx-10.16"
+  else
+    echo "Z3 support not configured for this platform (uname=$(uname))"
+    return
+  fi
+  TMPFILE=$(mktemp)
+  rm "$TMPFILE"
+  mkdir -p "$TMPFILE"/
+  (
+    cd "$TMPFILE" || exit
+    curl -LOs "https://github.com/Z3Prover/z3/releases/download/z3-$Z3_VERSION/$Z3_PKG.zip"
+    unzip -q "$Z3_PKG.zip"
+    cp "$Z3_PKG/bin/z3" "${INSTALL_DIR}"
+    chmod +x "${INSTALL_DIR}z3"
+  )
+  rm -rf "$TMPFILE"
+}
+
+function install_cvc5 {
+  echo "Installing cvc5"
+  if command -v /usr/local/bin/cvc5 &>/dev/null; then
+    echo "cvc5 already exists at /usr/local/bin/cvc5"
+    echo "but this install will go to $${INSTALL_DIR}cvc5."
+    echo "you may want to remove the shared instance to avoid version confusion"
+  fi
+  if command -v "${INSTALL_DIR}cvc5" &>/dev/null && [[ "$("${INSTALL_DIR}cvc5" --version || true)" =~ .*${CVC5_VERSION}.* ]]; then
+     echo "cvc5 ${CVC5_VERSION} already installed"
+     return
+  fi
+  if [[ "$(uname)" == "Linux" ]]; then
+    CVC5_PKG="cvc5-Linux"
+  elif [[ "$(uname)" == "Darwin" ]]; then
+    CVC5_PKG="cvc5-macOS"
+  else
+    echo "cvc5 support not configured for this platform (uname=$(uname))"
+    return
+  fi
+  TMPFILE=$(mktemp)
+  rm "$TMPFILE"
+  mkdir -p "$TMPFILE"/
+  (
+    cd "$TMPFILE" || exit
+    curl -LOs "https://github.com/cvc5/cvc5/releases/download/cvc5-$CVC5_VERSION/$CVC5_PKG"
+    cp "$CVC5_PKG" "${INSTALL_DIR}cvc5"
+    chmod +x "${INSTALL_DIR}cvc5"
+  )
+  rm -rf "$TMPFILE"
+}
+
+function install_pkg {
+  package=$1
+  PACKAGE_MANAGER=$2
+  PRE_COMMAND=()
+  if [ "$(whoami)" != 'root' ]; then
+    PRE_COMMAND=(sudo)
+  fi
+  if command -v "$package" &>/dev/null; then
+    echo "$package is already installed"
+  else
+    echo "Installing ${package}."
+    if [[ "$PACKAGE_MANAGER" == "yum" ]]; then
+      "${PRE_COMMAND[@]}" yum install "${package}" -y
+    elif [[ "$PACKAGE_MANAGER" == "apt-get" ]]; then
+      "${PRE_COMMAND[@]}" apt-get install "${package}" --no-install-recommends -y
+      echo apt-get install result code: $?
+    elif [[ "$PACKAGE_MANAGER" == "pacman" ]]; then
+      "${PRE_COMMAND[@]}" pacman -Syu "$package" --noconfirm
+    elif [[ "$PACKAGE_MANAGER" == "apk" ]]; then
+      apk --update add --no-cache "${package}"
+    elif [[ "$PACKAGE_MANAGER" == "dnf" ]]; then
+      dnf install "$package"
+    elif [[ "$PACKAGE_MANAGER" == "brew" ]]; then
+      brew install "$package"
+    fi
+  fi
+}
+
+function welcome_message {
+cat <<EOF
+Welcome to Aptos!
+
+This script will download and install the necessary dependencies needed to
+build, test and inspect Aptos Core.
+
+Based on your selection, these tools will be included:
+EOF
+
+  if [[ "$INSTALL_PROVER" == "true" ]]; then
+cat <<EOF
+Move prover tools (since -y was provided):
```

Comment:
@rahxephon89 , perhaps, we don't need `-y` option? The purpose of running this script is to install the Prover dependency?

Reply:
Thanks for the review Junkil! If -yp is not specified, the script will install all dependencies by default.